pandas/types/cast.py

```diff
@@ -983,3 +984,60 @@ def find_common_type(types):
             return np.object
 
     return np.find_common_type(types, [])
+
+
+def try_cast(arr, take_fast_path, data, dtype=None, copy=False,
+             raise_cast_failure=False):
+    """ try casting the array using dtype. Used in _sanitize_array procedure
+    at Series initialization
+    """
+    from pandas.core.categorical import Categorical, CategoricalAccessor
+
+    # perf shortcut as this is the most common case
+    if take_fast_path:
+        if maybe_castable(arr) and not copy and dtype is None:
+            return arr
+
+    try:
+        subarr = maybe_cast_to_datetime(arr, dtype)
+        if not is_extension_type(subarr):
+            subarr = np.array(subarr, dtype=dtype, copy=copy)
+
+            if dtype is not None:
+                d2 = data
```

Comment:
you don't need any of this is_list_like stuff, by definition this is *already* an array